### PR TITLE
enhance development support on macOS ARM

### PR DESCRIPTION
**Pull request by kzantow**
This PR adds support for building and testing Syft using `make` on `macOS` with `ARM`-based processors, predominantly by upgrading base images which include `arm` support.

### CODE REVIEW

**Comment by spiffcs** [comment]
@tgerla I think you're the only other person with an M1 mac currently that I know that could test this. If anyone in the community has one and can validate these changes then that would be great!

File: test/integration/package_deduplication_test.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+//go:build !arm64
```
Comment (spiffcs) [inline test/integration/package_deduplication_test.go:1]: Nice!

File: test/integration/test-fixtures/image-java-no-main-package/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM jenkins:2.60.3
+FROM jenkins/jenkins:2.346.3-slim-jdk17
```
Comment (spiffcs) [inline test/integration/test-fixtures/image-java-no-main-package/Dockerfile:1]: Thanks for the update here - was this causing an issue on macOS ARM?

**Comment by spiffcs** [comment]
confirmed that these Makefile changes are still backwards compatible on non macOS ARM platforms and do not change development workflows besides enhancing compatibility

**Comment by tgerla** [comment]
This is working fine for me on my M1 Mac. The only thing that fails is the build step that generates a .deb. It looks like some sort of architecture mismatch. Maybe that amd64 deb build can be skipped on arm64 configurations?

(full build log: https://gist.github.com/tgerla/4e986e37eadcefeb1dea8742f25c8b45)

```
+ docker run --rm -v /var/run/docker.sock://var/run/docker.sock -v //Users/tgerla/git/anchore/syft:/src -v /var/folders/n2/7zlrwfws2q15lkc4pc8tfr080000gn/T/syft-acceptance-test-deb-XXXXXX.PhRt4woq:/var/folders/n2/7zlrwfws2q15lkc4pc8tfr080000gn/T/syft-acceptance-test-deb-XXXXXX.PhRt4woq -e SYFT_CHECK_FOR_APP_UPDATE=0 -w /src ubuntu:latest /bin/bash -x -c '            DEBIAN_FRONTEND=noninteractive apt install ./snapshot/syft_*_linux_amd64.deb -y &&             syft version &&             syft packages centos:8.2.2004 -vv -o json > /var/folders/n2/7zlrwfws2q15lkc4pc8tfr080000gn/T/syft-acceptance-test-deb-XXXXXX.PhRt4woq/acceptance-deb-centos-8.2.2004.json         '
+ DEBIAN_FRONTEND=noninteractive
+ apt install ./snapshot/syft_0.41.3-SNAPSHOT-aa82f55_linux_amd64.deb -y

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  syft:amd64
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/21.3 MB of archives.
After this operation, 60.0 MB of additional disk space will be used.
Get:1 /src/snapshot/syft_0.41.3-SNAPSHOT-aa82f55_linux_amd64.deb syft amd64 0.41.3~SNAPSHOT-aa82f55 [21.3 MB]
debconf: delaying package configuration, since apt-utils is not installed
dpkg: error processing archive /src/snapshot/syft_0.41.3-SNAPSHOT-aa82f55_linux_amd64.deb (--unpack):
 package architecture (amd64) does not match system (arm64)
Errors were encountered while processing:
 /src/snapshot/syft_0.41.3-SNAPSHOT-aa82f55_linux_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
+ cleanup
+ exit_code=100
+ rm -rf /var/folders/n2/7zlrwfws2q15lkc4pc8tfr080000gn/T/syft-acceptance-test-deb-XXXXXX.PhRt4woq
+ exit 100
make: *** [compare-test-deb-package-install] Error 100
```

File: test/integration/test-fixtures/image-java-no-main-package/Dockerfile (hunk 2 above)
Comment (kzantow) [reply test/integration/test-fixtures/image-java-no-main-package/Dockerfile:1]: Yes, there were no arm images prior to this version, or at least... this version works with our tests and provides arm images :)

**Comment by spiffcs** [comment]
I think all that's needed here is to resolve the Makefile conflicts. We can address @tgerla's comment in a separate PR since I think there is a non-obvious way to skip/fix that check on the m2

**Review by spiffcs — APPROVED** [review]

**Comment by spiffcs** [comment]
TBD - exclusion/fix of build to resolve: https://gist.github.com/tgerla/4e986e37eadcefeb1dea8742f25c8b45

**Pull request merged**